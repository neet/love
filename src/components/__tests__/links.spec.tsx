import { shallow } from 'enzyme';
import * as React from 'react';
import * as ReactTestRenderer from 'react-test-renderer';
import Links from '../links';

const links = [
  {
    name: 'Twitter',
    label: '@TheGodOfNeet',
    href: 'https://twitter.com/thegodofneet',
    fa: 'fab fa-twitter',
    color1: '#297ccd',
    color2: 'white',
    tag: 'social',
  },
  {
    name: 'Mastodon',
    label: '@neet@mastodon.social',
    href: 'https://mastodon.social/@neet',
    fa: 'fab fa-mastodon',
    color1: '#1981e0',
    color2: 'white',
    tag: 'social',
  },
  {
    name: 'Discord',
    label: 'neetshin#1400',
    copy: 'neetshin#1400',
    fa: 'fab fa-discord',
    color1: '#7289da',
    color2: 'white',
    tag: 'social',
  },
];

describe('Links', () => {
  it('renders a avatar component with given props', () => {
    const tree = ReactTestRenderer.create(
      <Links links={links} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
