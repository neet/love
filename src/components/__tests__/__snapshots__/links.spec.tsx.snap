// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renders a avatar component with given props 1`] = `
<div
  className="links section"
>
  <h3>
    Contact
  </h3>
  <ul
    className="links__list"
    role="listbox"
  >
    <li
      aria-posinset={1}
      aria-setsize={3}
      className="links__list-item"
      role="listitem"
    >
      <div
        className="link-button-wrapper"
      >
        <a
          className="link-button"
          data-name="Twitter"
          href="https://twitter.com/thegodofneet"
          onClick={[Function]}
          target="__blank"
          title="@TheGodOfNeet"
        >
          <div
            className="link-button__fa-wrapper"
            style={
              Object {
                "backgroundColor": "#297ccd",
                "color": "white",
              }
            }
          >
            <i
              aria-hidden={true}
              className="link-button__fa fab fa-twitter"
            />
          </div>
          <div
            className="link-button__text"
          >
            <h4
              className="link-button__name not-special-font"
            >
              Twitter
            </h4>
            <p
              className="link-button__label"
            >
              @TheGodOfNeet
            </p>
          </div>
        </a>
      </div>
    </li>
    <li
      aria-posinset={2}
      aria-setsize={3}
      className="links__list-item"
      role="listitem"
    >
      <div
        className="link-button-wrapper"
      >
        <a
          className="link-button"
          data-name="Mastodon"
          href="https://mastodon.social/@neet"
          onClick={[Function]}
          target="__blank"
          title="@neet@mastodon.social"
        >
          <div
            className="link-button__fa-wrapper"
            style={
              Object {
                "backgroundColor": "#1981e0",
                "color": "white",
              }
            }
          >
            <i
              aria-hidden={true}
              className="link-button__fa fab fa-mastodon"
            />
          </div>
          <div
            className="link-button__text"
          >
            <h4
              className="link-button__name not-special-font"
            >
              Mastodon
            </h4>
            <p
              className="link-button__label"
            >
              @neet@mastodon.social
            </p>
          </div>
        </a>
      </div>
    </li>
    <li
      aria-posinset={3}
      aria-setsize={3}
      className="links__list-item"
      role="listitem"
    >
      <div
        className="link-button-wrapper"
      >
        <a
          className="link-button"
          data-name="Discord"
          onClick={[Function]}
          target="__blank"
          title="neetshin#1400"
        >
          <div
            className="link-button__fa-wrapper"
            style={
              Object {
                "backgroundColor": "#7289da",
                "color": "white",
              }
            }
          >
            <i
              aria-hidden={true}
              className="link-button__fa fab fa-discord"
            />
          </div>
          <div
            className="link-button__text"
          >
            <h4
              className="link-button__name not-special-font"
            >
              Discord
            </h4>
            <p
              className="link-button__label"
            >
              neetshin#1400
            </p>
          </div>
        </a>
      </div>
    </li>
  </ul>
  <h3>
    Development
  </h3>
  <ul
    className="links__list"
    role="listbox"
  />
  <h3>
    Donation
  </h3>
  <ul
    className="links__list"
    role="listbox"
  />
</div>
`;
